tests/PalindromeTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+use Interview\Solutions\Strings\Reverse\Reverse;
+use PHPUnit\Framework\TestCase;
+
+final class PalindromeTest extends TestCase {
+
+	/** @var string */
+	protected $palindrome;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->palindrome = $this->palindromize( function() {
+			// empower algorithm robustness by rapidiously leveraging core functionality to architect large number generated in the wild
+			return ( new DateTime() )->diff( new DateTime('0001-01-01') )->format('%a');
+		});
+	}
+
+	/**
+	 * @throws \Interview\Solutions\Strings\Reverse\Exceptions\InvalidAlgorithmException
+	 */
+	public function testDoesPalindromeReverse() : void {
+		$this->assertEquals( $this->palindrome, Reverse::reverseString( $this->palindrome ) );
+	}
+
+	/**
+	 * @throws \Interview\Solutions\Strings\Reverse\Exceptions\InvalidAlgorithmException
+	 */
+	public function testDoesPalindromeReverseTheReversal() : void {
+		$reversed_reversal = Reverse::reverseString(
+			Reverse::reverseString(
+				$this->palindrome
+			)
+		);
+
+		$this->assertEquals( $this->palindrome, $reversed_reversal );
+	}
+
+	/**
+	 * Palindromist helper that palindromically palindromizes a palindrome of n length
+	 *
+	 * @param callable $closure requested palindrome's length
+	 *
+	 * @return string
+	 * @todo move to phpunit fixture
+	 */
+	public function palindromize(callable $closure) : string {
+		$length = $closure();
+		$palindrome = '';
+
+		$list = new SplDoublyLinkedList();
+
+		for($i = 0; $i < $length; $i++) {
+			$list->push( $this->rand() );
+		}
+
+		for( $list->rewind(); $list->valid(); $list->next() ) {
+			$palindrome .= $list->current();
+		}
+
+		$list->setIteratorMode( SplDoublyLinkedList::IT_MODE_LIFO );
+
+		for( $list->rewind(); $list->valid(); $list->next() ) {
+			$palindrome .= $list->current();
+		}
+
+		return $palindrome;
+	}
+
+	public function rand() {
+		$chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
```

Comment:
Should refactor this into a fixture since multiple tests use this logic.. Now, [fixtures are a known anti-pattern according to phpunit](https://phpunit.de/manual/current/en/fixtures.html) and all, but they don't know what they're talking about. I'm the one with all the industry certifications listed in my Outlook signature after all.

Reply:
Tech debt is an important part of enterprise software, so go ahead and put it in something inconvenient like a spreadsheet and place it somewhere where you can be sure nobody will find it.

PR Approved.